src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/PerformanceTrackerService.cs

```diff
@@ -61,6 +62,37 @@ private PerformanceQueue GetQueue(bool forSpanAnalysis)
 
         public void AddSnapshot(IEnumerable<AnalyzerPerformanceInfo> snapshot, int unitCount, bool forSpanAnalysis)
         {
+            foreach (var perfInfo in snapshot)
+            {
+                const string Name = nameof(Name);
+                const string Delay = nameof(Delay);
+                const string IndividualTimes = nameof(IndividualTimes);
+                const int PerformAnalysisTelemetryDelay = 250;
+
+                var delay = (int)perfInfo.TimeSpan.TotalMilliseconds;
+
+                TelemetryLogging.LogAggregated(FunctionId.PerformAnalysis_Summary, KeyValueLogMessage.Create(m =>
+                {
+                    m[Name] = IndividualTimes;
+                    m[Delay] = delay;
+                }));
+
+                if (delay > PerformAnalysisTelemetryDelay)
+                {
+                    const string AnalyzerId = nameof(AnalyzerId);
+
+                    var analyzerId = perfInfo.BuiltIn ? perfInfo.AnalyzerId : perfInfo.BuiltIn.GetHashCode().ToString();
+
+                    var logMessage = KeyValueLogMessage.Create(m =>
+                    {
+                        m[AnalyzerId] = analyzerId;
+                        m[Delay] = delay;
+                    });
+
+                    TelemetryLogging.Log(FunctionId.PerformAnalysis_Delay, logMessage);
```

Comment:
We should account for the `forSpanAnalysis` parameter which indicates if this is lightbulb/partial document analysis or full document analysis.

Reply:
It seems we are logging a telemetry event for each diagnostic analyzer snapshot added in OOP. This is fine as of now as #67315 turned off telemetry logging for background analysis, we now do so only for explicit lightbulb invocation (Ctrl + Dot) code path. However, if we did end up logging this on a different path where lot of snapshots are being added, then this might end up with the same issue.

Reply:
Added forSpanAnalysis info to telemetry message in commit 4. Just to clarify, the *individual* telemetry event is only sent if the execution time of this analyzer exceeded the time threshold (250 ms), so hopefully there shouldn't be many that take that long (at least, I don't see many of them when operating on the languageparser.cs file in the Roslyn sln). Let me know if you would like additional information sent or a change in the threshold or even a filter to prevent some notifications from being sent.